create_builder.go

```diff
@@ -121,23 +136,48 @@ func (c *Client) CreateBuilder(ctx context.Context, opts CreateBuilderOptions) e
 	}
 
 	for _, pkg := range opts.Config.Buildpacks.Packages() {
-		locatorType, err := buildpack.GetLocatorType(pkg.ImageName, []dist.BuildpackInfo{})
+		locater := pkg.ImageName
+		if locater == "" && pkg.URI != "" {
+			locater = pkg.URI
+		}
+
+		locatorType, err := buildpack.GetLocatorType(locater, []dist.BuildpackInfo{})
 		if err != nil {
 			return err
 		}
 
+		var bps []dist.Buildpack
 		switch locatorType {
 		case buildpack.PackageLocator:
 			mainBP, depBPs, err := extractPackagedBuildpacks(ctx, pkg.ImageName, c.imageFetcher, opts.Publish, opts.NoPull)
 			if err != nil {
 				return err
 			}
 
-			for _, bp := range append([]dist.Buildpack{mainBP}, depBPs...) {
-				bldr.AddBuildpack(bp)
+			bps = append([]dist.Buildpack{mainBP}, depBPs...)
+		case buildpack.URILocator:
+			err := ensureBPSupport(pkg.URI)
+			if err != nil {
+				return err
+			}
+
+			blob, err := c.downloader.Download(ctx, pkg.URI)
```

Comment:
Don't we already have the blob from line 103? It would be nice not to have to download it again.